main.js

```diff
@@ -241,87 +284,56 @@ function createPopup(index) {
+  const projectCardButtons = document.querySelectorAll('.project_card_button');
+
+  for (let i = 0; i < projectCardButtons.length; i += 1) {
+    ((index) => {
+      projectCardButtons[index].addEventListener('click', () => {
+        createPopup(index);
+        const popupWindow = document.getElementsByClassName('popup__window')[index];
+        popupWindow.style.display = 'block';
+      });
+    })(i);
   }
+});
 
-  const projectCardButton = document.createElement('button');
-  projectCardButton.classList.add('project_card_button');
-  projectCardButton.setAttribute('type', 'button');
-  projectCardButton.innerText = 'See Project';
-  projectCardWrapper.appendChild(projectCardButton);
+// Form Validation //
 
-  projectWrapper.appendChild(projectCard);
+/* ------- SAVE INPUT VALUES TO LOCAL STORAGE -------*/
+const contactForm = document.querySelector('.contact__form');
 
-  projectCardButton.addEventListener('click', () => createPopup(i));
-}
+const saveFormData = () => {
+  const formData = {
+    fullName: document.getElementById('fullname').value,
+    email: document.getElementById('email').value,
+    message: document.getElementById('message').value,
+  };
 
-// Form Validation //
+  localStorage.setItem('formData', JSON.stringify(formData));
+};
 
-const fullName = document.getElementById('fullname');
-const email = document.getElementById('email');
+/* ------- GET INPUT VALUES FROM LOCAL STORAGE -------*/
+const loadFormData = () => {
+  const storedFormData = localStorage.getItem('formData');
 
-fullName.removeAttribute('required');
+  if (storedFormData !== null) {
+    const formData = JSON.parse(storedFormData);
+    document.getElementById('fullname').value = formData.fullName;
+    document.getElementById('email').value = formData.email;
+    document.getElementById('message').value = formData.message;
+  }
+};
 
-function errorMessage() {
-  const errorWindow = document.createElement('div');
-  errorWindow.setAttribute('id', 'error-window');
-  const errorMsg = document.createElement('div');
-  errorMsg.setAttribute('id', 'error-message');
-  errorMsg.setAttribute('class', 'error-message');
-  const errorMsgText = document.createTextNode(
-    'Your email should have only lowercase letters',
-  );
-  errorMsg.appendChild(errorMsgText);
-  errorWindow.appendChild(errorMsg);
-  email.after(errorWindow);
-}
+/* ------- SAVE DATA ON SUBMIT -------*/
 
-email.addEventListener('input', () => {
-  const emailValue = email.value;
-  const contactForm = document.getElementsByClassName('contact__form')[0];
-  const emailError = document.getElementById('error-message');
-  if (emailValue !== emailValue.toLowerCase()) {
-    if (contactForm.contains(emailError) === false) {
-      errorMessage();
-    }
-  } else if (document.body.contains(emailError) === true) {
-    emailError.remove();
-  }
+contactForm.addEventListener('submit', (e) => {
+  e.preventDefault();
+  saveFormData();
 });
```

Comment:
- [ ] As mentioned by the previous reviewer, you're saving the data on `submit` event which is not following the project requirement. 
  It stated in the project requirement that the data is saved to the the `localStorage` whenever the user changes the content of any input(see Image). Therefore I suggest you save the data on `change` event to follow the project requirement
![Screenshot from 2023-07-28 22-20-54](https://github.com/mateoburvi/Portfolio/assets/91142432/d88692f6-edb8-4ca7-b2a3-115666c0cb0d)

**HIT: Your code could look like this**
```js
contactForm.addEventListener('change', (e) => {
  e.preventDefault();
  saveFormData();
});
```